components/spire-agent/Makefile

```diff
@@ -0,0 +1,32 @@
+TSI_VERSION=$(shell cat ../../tsi-version.txt)
+GIT_COMMIT_SHA="$(shell git rev-parse --short HEAD 2>/dev/null)"
+GIT_REMOTE_URL="$(shell git config --get remote.origin.url 2>/dev/null)"
+BUILD_DATE="$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")"
+BINARY_NAME="spire-agent"
+REPO ?= tsidentity
+IMAGE := $(REPO)/$(BINARY_NAME):$(GIT_COMMIT_SHA)
+MUTABLE_IMAGE := $(REPO)/$(BINARY_NAME):$(TSI_VERSION)
+
+all: docker timestamp
+
+allpush: docker docker-push timestamp
+
+fastpush: fast docker-push timestamp
+
+fast:
+	docker build -t $(IMAGE) .
+	docker tag $(IMAGE) $(MUTABLE_IMAGE)
+	date
+
+docker:
```

Comment:
would like to see if rebuilding the agent is necessary, not sure if just adding args in the yaml is good enough?